pyfixest/estimation/feiv_.py

```diff
@@ -193,3 +206,225 @@ def get_fit(self) -> None:
         # Compute bread matrix
         D = np.linalg.inv(self._tXZ @ self._tZZinv @ self._tZX)
         self._bread = H.T @ D @ H
+
+    def first_stage(self) -> None:
+        """Implement First stage regression."""
+        from pyfixest.estimation.estimation import feols
+
+        #  Start First Stage
+        _X = self._X
+        _Z = self._Z
+        _Y = self._Y
+
+        # Store names of instruments from Z matrix
+        self._non_exo_instruments = list(set(self._coefnames_z) - set(self._coefnames))
+
+        # Prepare your data
+        # Select the columns from self._data that match the variable names
+        data = pd.DataFrame(self._Z_1st_stage, columns=self._coefnames_z)
+
+        # Store instrument variable matrix for futyre use
+        _Z_iv = data[self._non_exo_instruments]
+        self._Z_iv = _Z_iv.to_numpy()
+        # Dynamically create the formula string
+        # Extract names of fixed effect
+
+        data["Y"] = self._endogvar_1st_stage
+        independent_vars = " + ".join(
+            data.columns[:-1]
+        )  # All columns except the last one ('Y')
+
+        # Set fix effects, cluster, and weight options to be passed to feols.
+
+        if self._has_fixef:
+            FE_vars = self._fixef.split("+")
+            data[FE_vars] = self._data[FE_vars]
+            formula = f"Y ~ {independent_vars} | {self._fixef}"
+        else:
+            formula = f"Y ~ {independent_vars}"
+
+        # Type hint to reflect that vcov_detail can be either a dict or a str
+        vcov_detail: Union[dict[str, str], str]
+
+        if self._is_clustered:
+            a = self._clustervar[0]
+            vcov_detail = {self._vcov_type_detail: a}
+            data[a] = self._data[a]
+        else:
+            vcov_detail = self._vcov_type_detail
+
+        if self._has_weights:
+            data["weights"] = self._weights
+            weight_detail = "weights"
+        else:
+            weight_detail = None
+
+        # Do first stage regression
+
+        model1 = feols(
+            fml=formula,
+            data=data,
+            vcov=vcov_detail,
+            weights=weight_detail,
+            collin_tol=1e-10,
+        )
+        # Ensure model1 is of type Feols
+        if isinstance(model1, Feols):
```

Comment:
Would it make sense to call `IV_Diag` within the first stage? E.g. here we could add a line 

```python 
self.IV_diag()
```
Motivation is that I assume that almost all users who run an IV regression will be interested in obtaining those statistics. 

Additionally, we could add a function argument `statistics` to `IV_Diag` which allows users to select **additional** statistics post estimation that we will add in the future. E.g. we'd have smth as 

```python 
def IV_Diag(self, statistics: Optional[list[str]] = None): 
   # default
   iv_diag_statistics = ["effective_f", "f_stat"]
   if not statistics: 
      iv_diag_statistics += statistics
```

Reply:
Oh yes, you are right. I should have put `IV_Diag` to `first_stage` method to make it a default option. Thanks for pointing out :)